components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/internal/TokenGeneratorRegistrationComponent.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.appmgt.gateway.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.carbon.appmgt.impl.AppManagerConfiguration;
+import org.wso2.carbon.appmgt.impl.AppManagerConfigurationService;
+import org.wso2.carbon.appmgt.impl.token.JWTGenerator;
+import org.wso2.carbon.appmgt.impl.token.TokenGenerator;
+
+/**
+ *
+ * Registers the default token generator. This is the internal JWT generator.
+ *
+ * @scr.component name="org.wso2.appmgt.impl.services.gateway.tokenGenerator" immediate="true"
+ * @scr.reference name="app.manager.config.service"
+ * interface="org.wso2.carbon.appmgt.impl.AppManagerConfigurationService"
+ * policy="dynamic" bind="setAppManagerConfigurationService" unbind="unsetAppManagerConfigurationService"
```

Comment:
Cardinality is not mandatory for OSGI declarative service but better to define a proper value.